src/pass/rematerialization.cc

```diff
@@ -592,6 +592,11 @@ class Rematerializer : public ExprMutator {
     static const auto reshape_op = Op::Get("raf.op.reshape");
 
     auto tensor_infos = tensor_infos_.GetTensorInfoFromLetVar(target_let_var);
+    if (tensor_infos.size() > 1) {
+      // TODO: Handle tuple. This happens for ops taking a tuple as an argument (e.g., concat).
+      // Ideally we should check each element in the tuple and correct their types.
+      return target_let_var;
+    }
```

Comment:
I'm wondering when this case will be triggered. Is it triggered every time we try to rematerialize a tuple?

Reply:
No. As the comment indicates, this happens when an argument is a tuple. For example:

```
let %a1 = (%x, %y);
let %a2 = concat(%a1, 0);
```

In this case remat pass will visit `CorrectType(%a1)`.

Reply:
I see. So it happens when we try to rematerialize the `concat`. The `Rematerialize()` function will visit the arguments where `a1` is a tuple.

Reply:
Exactly.